drivers: spi: Migrate spi_context_{complete,lock} usage

stephanosio opened the pull request:
This series updates the SPI driver to reflect the `spi_context_complete` and `spi_context_lock` function signature changes introduced in the commit https://github.com/zephyrproject-rtos/zephyr/commit/4c20403629df1ae6a58d37a7a5bd73d4698cc11a (https://github.com/zephyrproject-rtos/zephyr/pull/49180 forgot to migrate these).

Fixes various `error: too few arguments to function 'spi_context_complete'` errors seen in the CI.

stephanosio committed:
drivers: spi: gd32: Migrate spi_context_{complete,lock} usage

This commit updates the GD32 SPI driver to reflect the
`spi_context_complete` and `spi_context_lock` function signature
changes introduced in the commit
4c20403629df1ae6a58d37a7a5bd73d4698cc11a.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

stephanosio committed:
drivers: spi: gecko: Migrate spi_context_complete usage

This commit updates the Gecko SPI driver to reflect the
`spi_context_complete` function signature change introduced in the
commit 4c20403629df1ae6a58d37a7a5bd73d4698cc11a.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

stephanosio committed:
drivers: spi: litespi: Migrate spi_context_complete usage

This commit updates the LiteSPI SPI driver to reflect the
`spi_context_complete` function signature change introduced in the
commit 4c20403629df1ae6a58d37a7a5bd73d4698cc11a.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

stephanosio committed:
drivers: spi: mchp_mss_qspi: Migrate spi_context_{complete,lock} usage

This commit updates the Microchip MSS QSPI SPI driver to reflect the
`spi_context_complete` and `spi_context_lock` function signature
changes introduced in the commit
4c20403629df1ae6a58d37a7a5bd73d4698cc11a.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

stephanosio committed:
drivers: spi: sifive: Migrate spi_context_{complete,lock} usage

This commit updates the SiFive SPI driver to reflect the
`spi_context_complete` and `spi_context_lock` function signature
changes introduced in the commit
4c20403629df1ae6a58d37a7a5bd73d4698cc11a.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

henrikbrixandersen commented:
Looks like you're missing at least psoc6:
```

FAILED: zephyr/drivers/spi/CMakeFiles/drivers__spi.dir/spi_psoc6.c.obj 
ccache /opt/toolchains/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DCY8C6347BZI_BLD53 -DKERNEL -DTC_RUNID=d45fd1cce53baf9f8a47f41ef3f7fed6 -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../../../../../../../include -Izephyr/include/generated -I../../../../../../../soc/arm/cypress/psoc6 -I../../../../../../../soc/arm/cypress/psoc6/. -I../../../../../../../soc/arm/cypress/common/. -I../../../../../../../subsys/testsuite/include -I../../../../../../../subsys/testsuite/ztest/include -I/__w/zephyr/modules/hal/cmsis/CMSIS/Core/Include -I/__w/zephyr/modules/hal/infineon/core-lib/include -I/__w/zephyr/modules/hal/infineon/mtb-pdl-cat1/drivers/include -I/__w/zephyr/modules/hal/infineon/mtb-pdl-cat1/devices/COMPONENT_CAT1A/include -I/__w/zephyr/modules/hal/infineon/mtb-pdl-cat1/devices/COMPONENT_CAT1A/include/ip -I/__w/zephyr/modules/hal/infineon/mtb-pdl-cat1/devices/COMPONENT_CAT1A/templates/COMPONENT_MTB -isystem ../../../../../../../lib/libc/minimal/include -isystem /opt/toolchains/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/include -isystem /opt/toolchains/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/include-fixed -fno-strict-aliasing -Os -imacros /__w/zephyr/zephyr/twister-out/cy8ckit_062_ble_m0@0.0.0/tests/drivers/can/api/drivers.can.api.mcp2515/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m0plus -mthumb -mabi=aapcs -mfp16-format=ieee --sysroot=/opt/toolchains/zephyr-sdk-0.15.0/arm-zephyr-eabi/arm-zephyr-eabi -imacros /__w/zephyr/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -Werror -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/__w/zephyr/zephyr/tests/drivers/can/api=CMAKE_SOURCE_DIR -fmacro-prefix-map=/__w/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/__w/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -MD -MT zephyr/drivers/spi/CMakeFiles/drivers__spi.dir/spi_psoc6.c.obj -MF zephyr/drivers/spi/CMakeFiles/drivers__spi.dir/spi_psoc6.c.obj.d -o zephyr/drivers/spi/CMakeFiles/drivers__spi.dir/spi_psoc6.c.obj -c /__w/zephyr/zephyr/drivers/spi/spi_psoc6.c
/__w/zephyr/zephyr/drivers/spi/spi_psoc6.c: In function 'spi_psoc6_transceive_sync':
/__w/zephyr/zephyr/drivers/spi/spi_psoc6.c:347:16: error: too few arguments to function 'spi_psoc6_transceive'
  347 |         return spi_psoc6_transceive(dev, spi_cfg, tx_bufs,
      |                ^~~~~~~~~~~~~~~~~~~~
/__w/zephyr/zephyr/drivers/spi/spi_psoc6.c:294:12: note: declared here
  294 | static int spi_psoc6_transceive(const struct device *dev,
      |            ^~~~~~~~~~~~~~~~~~~~
/__w/zephyr/zephyr/drivers/spi/spi_psoc6.c:349:1: error: control reaches end of non-void function [-Werror=return-type]
  349 | }
      | ^
```

stephanosio commented:
> Looks like you're missing at least psoc6

That issue is unrelated to the issue this PR is fixing.

henrikbrixandersen commented:
> That issue is unrelated to the issue this PR is fixing.

How so? The change in number of function arguments to `spi_psoc6_transceive()` was introduced in 4c20403629df1ae6a58d37a7a5bd73d4698cc11a.

stephanosio committed:
drivers: spi: psoc6: Fix spi_psoc6_transceive calls

The commit 4c20403629df1ae6a58d37a7a5bd73d4698cc11a updated the
`spi_psoc6_transceive` function signature but forgot to update its
usages.

This commit updates the `spi_psoc6_transceive` function invocations
to properly provide the callback and user data parameters.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

stephanosio commented:
Reworked this PR to fix all problems introduced in the commit https://github.com/zephyrproject-rtos/zephyr/commit/4c20403629df1ae6a58d37a7a5bd73d4698cc11a.
